Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,263 @@
+        /// <param name="rules">The rules used to validate input schemas is only allowed in string or JSchema type</param>
+        /// <param name="validateAsDictionary">True if the custom rules is to handle the whole dictionary</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaAsString(string schema, object rules, bool validateAsDictionary, out IList<ValidationError> errorList)
+        {
+            List<JObject> schemaList = ConvertStringToJObjectList(schema, validateAsDictionary);
+
+            return Validate(schemaList, rules, out errorList);
+        }
+
+        /// <summary>
+        /// Validates single or multiple schemas from input path with the given rules.        
+        /// </summary>
+        /// <param name="path">The path that contains a schema file</param>
+        /// <param name="rules">The rules used to validate input schemas is only allowed in string or JSchema type</param>
+        /// <param name="validateAsDictionary">True if the custom rules is to handle the whole dictionary</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaFromPath(string path, object rules, bool validateAsDictionary, out IList<ValidationError> errorList)
+        {
+            List<JObject> schemas = GetSchemaFromPath(path, validateAsDictionary);
+
+            return Validate(schemas, rules, out errorList);
+        }
+
+        /// <summary>
+        /// Validates single or multiple schemas from input path with the given rules.
+        /// </summary>
+        /// <param name="path">The path that contains a schemas directory</param>
+        /// <param name="rules">The rules used to validate input schemas is only allowed in string or JSchema type</param>
+        /// <param name="validateAsDictionary">True if the custom rules is to handle the whole dictionary</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaFromDirectory(string path, object rules, bool validateAsDictionary, out IList<ValidationError> errorList)
+        {
+            List<JObject> schemas = GetAllSchemasInDirectory(path, validateAsDictionary);
+
+            return Validate(schemas, rules, out errorList);
+        }
+
+
+        private static List<JObject> ConvertDictionaryToJObjectList(Dictionary<string, ForgeTree> schemas, bool validateAsDictionary)
+        {
+            List<JObject> schemaList = new List<JObject>();
+
+            if (validateAsDictionary)
+            {
+                schemaList.Add(SerializeToJObject(schemas));
+            }
+            else
+            {
+                foreach (ForgeTree item in schemas.Values)
```

Comment:
>item [](http://example.com/codeflow?start=35&length=4)

Generally we want to name variables with their real names. "forgeTree" in this case.